purchase_delivery_split_date/models/purchase.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2014-2016 Numérigraphe SARL
+# Copyright 2017 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import datetime
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+
+import logging
+from itertools import groupby
+from odoo import models, api
+
+_logger = logging.getLogger(__name__)
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    @api.model
+    def _get_group_keys(self, order, line, picking=False):
+        """Define the key that will be used to group. The key should be
+        defined as a tuple of dictionaries, with each element containing a
+        dictionary element with the field that you want to group by. This
+        method is designed for extensibility, so that other modules can add
+        additional keys or replace them by others."""
+        date = datetime.strptime(
+            line.date_planned, DEFAULT_SERVER_DATETIME_FORMAT)
+        key = ({'date_planned': date.date()},)
+        return key
+
+    @api.model
+    def _first_picking_copy_vals(self, key, lines):
+        """The data to be copied to new pickings is updated with data from the
+        grouping key.  This method is designed for extensibility, so that
+        other modules can store more data based on new keys."""
+        vals = {'move_lines': []}
+        for key_element in key:
+            if 'date_planned' in key_element.keys():
+                vals['date'] = key_element['date_planned']
+        return vals
+
+    @api.multi
+    def _create_stock_moves(self, picking):
+        """Group the receptions in one picking per group key"""
+        moves = self.env['stock.move']
+        # Group the order lines by group key
+        order_lines = sorted(self, key=lambda l: l.order_id)
+        date_groups = groupby(order_lines, lambda l: self._get_group_keys(
+            l.order_id, l, picking=picking))
+
+        # If a picking is provided, use it for the first group only
+        if picking:
+            first_picking = picking
+            key, lines = next(date_groups)
+            po_lines = self.env['purchase.order.line']
+            for line in list(lines):
+                po_lines += line
+            picking._update_picking_from_group_key(key)
+            moves += super(PurchaseOrderLine, po_lines)._create_stock_moves(
+                first_picking)
+        else:
+            first_picking = False
```

Comment:
This can be put before the if statement. Saves some time trying to make the coverage 100%.